src/mod/l4_thruster.c

```diff
@@ -0,0 +1,539 @@
+/*
```

Comment:
I gave you general review on the thruster globally with mail. So for having a faster pull request to develop I mainly skip this part(s) for now. I assume it works and does not block mainloop and other functionality. 
We can make cleanup/rerfactorings for memory or other optimizations later.